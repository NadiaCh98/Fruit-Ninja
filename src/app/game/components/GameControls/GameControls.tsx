import React, { HTMLAttributes, memo } from 'react';
import styled from 'styled-components';

const StyledGameControls = styled.div`
  position: absolute;
  z-index: 2;
  padding: 2rem;
`;

export const GameControls: React.FC<HTMLAttributes<unknown>> = memo(
  ({ children, style }) => {
    return <StyledGameControls style={style}>{children}</StyledGameControls>;
  }
);

GameControls.displayName = 'GameControl';
