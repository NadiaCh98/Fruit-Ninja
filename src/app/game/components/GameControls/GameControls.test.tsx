import React from 'react';
import { render } from '@testing-library/react';
import { GameControls } from './GameControls';

describe('Game Controls', () => {
  it('has a correct template #1', () => {
    const { container } = render(<GameControls />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('has a correct template #2', () => {
    const { container } = render(<GameControls>Hello</GameControls>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
