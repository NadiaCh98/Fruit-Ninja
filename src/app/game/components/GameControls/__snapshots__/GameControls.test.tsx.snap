// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Controls has a correct template #1 1`] = `
<div
  class="sc-bdnxRM lhbDim"
/>
`;

exports[`Game Controls has a correct template #2 1`] = `
<div
  class="sc-bdnxRM lhbDim"
>
  Hello
</div>
`;
